src/InterpolationPoint_TEST.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include <ignition/math/detail/InterpolationPoint.hh>
+
+#include <gtest/gtest.h>
+
+using namespace ignition;
+using namespace math;
+
+TEST(Interpolation, LinearInterpolate)
+{
+  std::vector<double> vec{0, 2};
+
+  InterpolationPoint1D<double> first {0.0, 0};
+  InterpolationPoint1D<double> last {1.0, 1};
+
+  EXPECT_NEAR(LinearInterpolate(first, last, vec, 0.0), 0, 1e-3);
+  EXPECT_NEAR(LinearInterpolate(first, last, vec, 1.0), 2, 1e-3);
+  EXPECT_NEAR(LinearInterpolate(first, last, vec, 0.5), 1, 1e-3);
```

Comment:
@arjo129 consider adding tests for when (a) `first` and `last` are reversed, (b) values in `vec` are all the same, and (c) `first` and `last` are the same point.

Reply:
One final note about (c). If we intend to rely on `AxisIndex` to return fewer points for this situation to never arise, document it as a precondition and `assert` it.